packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -418,26 +422,30 @@ class _DragAvatar<T> extends Drag {
     _entry.markNeedsBuild();
     HitTestResult result = new HitTestResult();
     WidgetFlutterBinding.instance.hitTest(result, globalPosition + feedbackOffset);
-    _DragTargetState<T> target = _getDragTarget(result.path);
-    if (target == _activeTarget)
+    List<_DragTargetState<T>> targets = _getDragTargets(result.path);
+    if (targets.contains(_activeTarget))
       return;
     if (_activeTarget != null)
       _activeTarget.didLeave(data);
-    _activeTarget = target;
+    _activeTarget = targets.firstWhere(
+      (_DragTargetState<T> target) => target.willAccept(data),
+      orElse: () => targets.isEmpty ? null : targets.first
+    );
     _activeTargetWillAcceptDrop = _activeTarget != null && _activeTarget.didEnter(data);
```

Comment:
I think what the semantics should be this:

- call didEnter on every node until one returns true
- remember all the nodes that said no, and the one that said yes, if any
- next time, if the list is identical up to the node that said yes, then early exit
- otherwise, call didExit on all the nodes and try again.